url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/http/HttpRequestHeader.java

```diff
@@ -0,0 +1,85 @@
+package net.hetimatan.util.http;
+
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import net.hetimatan.io.file.MarkableFileReader;
+import net.hetimatan.io.file.MarkableReader;
+import net.hetimatan.io.file.MarkableReaderHelper;
+import net.hetimatan.io.filen.CashKyoroFile;
+
+// KEY ":" VALUE CRLF
+public class HttpRequestHeader extends HttpObject {
+
+	private String mKey = "";
+	private String mValue = "";
+	public static final String SEPARATOR = ":";
+	public static final String HEADER_CONTENT_LENGTH = "Content-Length";
+	public static final String HEADER_CONTENT_TyPE = "Content-Type";
+	public static final String HEADER_LOCATION = "Location";
+	public static final String HEADER_HOST = "Host";
+
+	public HttpRequestHeader(String key, String value) {
+		mKey = key;
+		mValue = value;
+	}
+
+	@Override
+	public void encode(OutputStream output) throws IOException {
+		output.write(mKey.getBytes());
+		output.write(SEPARATOR.getBytes());
+		output.write(" ".getBytes());
+		output.write(mValue.getBytes());
+		output.write(CRLF.getBytes());
+	}
+
+	public String getKey() {
+		return mKey;
+	}
+
+	public String getValue() {
+		return mValue;
+	}
+
+	public static HttpRequestHeader decode(String path) throws IOException {
+		CashKyoroFile vFile = new CashKyoroFile(512, 2);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *vFile*. Currently, there are execution paths that do not contain closure statements, for example, when CashKyoroFile.addChunk throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).